### DISCUSSION TITLE
Is it possible to use Directus layouts in custom modules?

### DISCUSSION THREAD

**Discussion by goldenpath-ai** [directus/directus#15204] [Archive]
Hello,

is it possible to use both custom and Directus layouts in my custom modules? I know `useLayout` exists but I have not been able to find any documentation how to use it properly and couldn't figure it out through its uses in Directus code.

```
<template>
  <private-view title="Calendar">
    <div class="wrapper">something here</div>


   //Here I want to display the native calendar layout with a chosen collection. Is it possible somehow?
    <component
      :is="layoutWrapper"
      v-slot="{ layoutState }"
      :filter="''"
      :search="''"
      :collection="'Letter'"
    >
      <div>{{ layoutState }}</div>
    </component>
  </private-view>
</template>

<script lang="ts" setup>
import { useLayout } from "@directus/extensions-sdk";

import { ref } from "vue";

const id = ref("calendar");
const { layoutWrapper } = useLayout(id as any);
</script>

<style>
.wrapper {
  padding: 0 50px;
}
</style>
```

**Comment by rijkvanzanten** [comment]
All the core layouts are globally registered components under the name `layout-{layout id}` here:

https://github.com/directus/directus/blob/add851ac16ac0e24363a2f4041b4eb08584b59d5/app/src/layouts/register.ts#L28-L33

so you can use 

```html
<layout-calendar />
```

in your own vue component to render the calendar layout 🙂

**Reply by goldenpath-ai** [reply]
I've tried that, however I'm getting an error `t.createCalendar is not a function`. I'm on latest directus version.

**Reply by goldenpath-ai** [reply]
Any ideas what might be incorrect here? @rijkvanzanten

**Reply by rijkvanzanten** [reply]
No idea.. I've never tried rendering a layout in another layout, so not quite sure if/how that works in real life use

**Reply by michaeltoohig** [reply]
@filipkrayem any success on using the native layout? I'm looking for a working example.

**Reply by goldenpath-ai** [reply]
@michaeltoohig in my case I needed to use the calendar layout, which I achieved by adding it as a dependency of the module and then using the library itself. Didn't have any success using the actual native layout.

After working with directus and its custom modules for a few months, I advise you to stay as far away from them as you can if possible.
